pkg/provisioners/gcppubsub/controller/channel/names.go

```diff
@@ -14,30 +14,25 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package channel
 
 import (
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
 	"github.com/knative/eventing/pkg/provisioners/utils"
 
 	eventduck "github.com/knative/eventing/pkg/apis/duck/v1alpha1"
 )
 
-// Underscore is used as a separator between the namespace and name so the concatenated version
-// can't accidentally overlap.
+// Underscore is used as a separator so the concatenated version can't accidentally overlap.
 
-// GenerateTopicName generates the GCP PubSub Topic name given the Knative Channel's namespace and
-// name.
-// Note that it must be stable. The controller and dispatcher rely on this being generated
-// identically.
-func GenerateTopicName(channelNamespace, channelName string) string {
-	return utils.TopicName("_", channelNamespace, channelName)
+// generateTopicName generates the GCP PubSub Topic name given the Knative Channel.
+func generateTopicName(c *v1alpha1.Channel) string {
+	return utils.TopicName("_", c.Name, string(c.UID))
```

Comment:
perhaps other _CCP_s should do similarly ?

Reply:
I interpreted this comment as should all CCP's name the topic based on the UID? If that is the wrong interpretation, let me know.

It depends. I think the utility was first extracted from the `in-memory-channel`, which only requires name and namespace to be completely unique in time. I would guess that (default) Kafka and NATSS are the same, because they run completely inside the cluster. So only the `Channel`'s name and namespace are needed for uniqueness. 

GCP PubSub runs outside of the cluster so now needs to differentiate between a `Channel` with the same name and namespace in two different clusters. I happen to be using the UID here for that differentiation. The `Channel`'s name is added for humans that happen to look at it, but is not needed for uniqueness reasons. Note that Kafka and NATSS would have similar problems if multiple K8s clusters used the same Kafka/NATSS data store.